assets/js/platform-utils.js

```
@@ -0,0 +1,42 @@
+"use strict";
+
+/**
+ * Determines whether the user"s platform is macOS.
+ *
+ * @returns {boolean} True if the current platform is macOS; otherwise, false.
+ */
+function isMacPlatform() {
+  // First try modern User-Agent Client Hints API
+  if (navigator.userAgentData && navigator.userAgentData.platform) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)